internal/backends/linuxkms/renderer/sw.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+//! Delegate the rendering to the [`i_slint_core::software_renderer::SoftwareRenderer`]
+
+use i_slint_core::api::PhysicalSize as PhysicalWindowSize;
+use i_slint_core::platform::PlatformError;
+pub use i_slint_core::software_renderer::SoftwareRenderer;
+use i_slint_core::software_renderer::{PremultipliedRgbaColor, RepaintBufferType, TargetPixel};
+use std::rc::Rc;
+
+use crate::display::{Presenter, RenderingRotation};
+use crate::drmoutput::DrmOutput;
+
+pub struct SoftwareRendererAdapter {
+    renderer: SoftwareRenderer,
+    display: Rc<crate::display::swdisplay::SoftwareBufferDisplay>,
+    size: PhysicalWindowSize,
+}
+
+#[repr(transparent)]
+#[derive(Copy, Clone, bytemuck::Pod, bytemuck::Zeroable)]
+struct DumbBufferPixel(pub u32);
```

Comment:
I don't see an argb8pixel in our code base.

I see these:

 - Rgba8Pixel: TargetPixel is not implemented for it and it's the wrong order. (Softbuffer expects 0RGB)
 - PremultipliedRgbaColor: TargetPixel is implemented for it, but it's the wrong order.

This is basically the same as in commit 46ec7875ed08358b812e1722be96cbf52718773a . If you have a suggestion where it could be shared, then I'm all for it. Should I introduce a "softbuffer-pixel" feature in i-slint-core perhaps? I wonder in which module to place it.